activity/activity_IngestDigestToEndpoint.py

```
@@ -0,0 +1,377 @@
+        jats_file = None
+        expanded_bucket_name = (self.settings.publishing_buckets_prefix
+                                + self.settings.expanded_bucket)
+        try:
+            jats_file = download_article_xml(
+                self.settings, self.temp_dir, expanded_folder_name, expanded_bucket_name)
+        except Exception as exception:
+            self.logger.exception(
+                "Exception generating digest json for docx_file %s. Details: %s" %
+                (str(docx_file), str(exception)))
+        return jats_file
+
+    def digest_json(self, docx_file, jats_file=None, image_file=None, related=None):
+        "generate the digest json content from the docx file and other data"
+        json_content = None
+        try:
+            json_content = json_output.build_json(docx_file, self.temp_dir, self.digest_config,
+                                                  jats_file, image_file, related)
+        except Exception as exception:
+            self.logger.exception(
+                "Exception generating digest json for docx_file %s. Details: %s" %
+                (str(docx_file), str(exception)))
+        return json_content
+
+    def put_digest_to_endpoint(self, digest_id, digest_content, settings):
+        "handle issuing the PUT to the digest endpoint"
+        put_status = None
+        try:
+            status_code, response = digest_provider.put_digest(digest_id, digest_content, settings)
+            if status_code == 204:
+                put_status = True
+            else:
+                self.logger.error("PUT to digest endpoint status code  %s. Response: %s" %
+                                  str(status_code), str(response))
+        except Exception as exception:
+            self.logger.exception(
+                "Exception issuing PUT to the digest endpoint for digest_id %s. Details: %s" %
+                (str(digest_id), str(exception)))
+            put_status = False
```

same as before, once logging has been dealt with, can throw rather than returning a boolean